hcl/hcl.go

```diff
@@ -415,6 +454,33 @@ func validateGenerateHCLBlock(block *hclsyntax.Block) error {
 	return nil
 }
 
+func validateGenerateFileBlock(block *hclsyntax.Block) error {
+	if len(block.Labels) != 1 {
+		return errors.E(ErrTerramateSchema, block.OpenBraceRange,
+			"generate_file must have single label instead got %v",
+			block.Labels,
+		)
+	}
+	if block.Labels[0] == "" {
+		return errors.E(ErrTerramateSchema, block.OpenBraceRange,
+			"generate_file label can't be empty")
+	}
+	schema := &hcl.BodySchema{
+		Attributes: []hcl.AttributeSchema{
+			{
+				Name:     "content",
+				Required: true,
+			},
+		},
+	}
+
+	_, diags := block.Body.Content(schema)
```

Comment:
this gives me shivers :joy: 
When refactoring for the config loader I'd like to property parse everything ourselves.

Reply:
I was not that aware that the schema checks from the HCL lib were terrible, but I believe you :laughing: